### PR TITLE
add illinois similarity module

### LINKED ISSUE #185: core-utils readme: no mention of lazy initialization
Lazy Initialization was recently been added by @mssammon in #148 .
Unfortunately there is no mention of this in the readme: https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/tree/master/core-utilities 

This should carefully be explained, otherwise it would create much confusions.

### LINKED ISSUE #256: Rename nonDefaultRm in Annotator.java
This name is confusing, this could also be the default rm if we lazily initialize

### PR DESCRIPTION
**Pull request by mssammon**
and minor fixes: rename ResourceManager field in Annotator (closes #256) and updates core-utilities README to cover Annotator and lazy initialization (closes #185)

### CODE REVIEW

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/LbjTagger/NEWord.java [hunk 1]
```diff
@@ -107,6 +107,19 @@ public NEWord(Word w, NEWord p, String type) {
...
+    public NEWord() {
+        super("");
+        label = "";
+        form = "";
+        originalForm = "";
+        neLabel = "";
+        neTypeLevel1 = "";
+        parts = new String[0];
+    }
+
```
Comment (mssammon) [inline ner/src/main/java/edu/illinois/cs/cogcomp/ner/LbjTagger/NEWord.java]: Good catch.  This was the result of a painful 3-way merge with my fork. Removing it...

**Comment by mssammon** [comment]
@danyaljj  any idea why the semaphore log is shwing this? 
semaphore-log

**Comment by mssammon** [comment]
URL I followed from 'failed' link: https://semaphoreci.com/cogcomp/illinois-cogcomp-nlp/branches/pull-request-271/builds/5

**Comment by danyaljj** [comment]
@mssammon Downloading the resources. Restarted semaphore and it passed

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/LbjTagger/NEWord.java [hunk 2]
```diff
@@ -36,10 +32,9 @@
     public CharacteristicWords predictionConfidencesLevel1Classifier = null;
     public CharacteristicWords predictionConfidencesLevel2Classifier = null;
     public NamedEntity goldEntity = null;// if non-null it keeps the named entity the tagger
-                                         // annotated this word with
     /** This field stores the named entity type tag found in labeled data. */
     public String neLabel = null;
-
+                                         // annotated this word with
```
Comment (danyaljj) [inline ner/src/main/java/edu/illinois/cs/cogcomp/ner/LbjTagger/NEWord.java]: what is this comment?

**Comment by danyaljj** [comment]
This looks good. Merge this @mssammon?

**Comment by mssammon** [comment]
yes please...

**Pull request merged**